src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -151,6 +155,63 @@ static BoundNode unwrapListElement(BoundCollectionExpression node, BoundNode ele
             }
         }
 
+        // If we have something like `List<int> l = [.. someEnumerable]`
+        // try rewrite it using `Enumerable.ToList` member if possible
+        private bool TryRewriteSingleElementSpreadToList(BoundCollectionExpression node, TypeWithAnnotations listElementType, [NotNullWhen(true)] out BoundExpression? result)
+        {
+            result = null;
+
+            if (node.Elements is not [BoundCollectionExpressionSpreadElement singleSpread])
+            {
+                return false;
+            }
+
+            if (!TryGetWellKnownTypeMember(node.Syntax, WellKnownMember.System_Linq_Enumerable__ToList, out MethodSymbol? toListGeneric, isOptional: true))
+            {
+                return false;
+            }
+
+            var toListOfElementType = toListGeneric.Construct([listElementType]);
+
+            Debug.Assert(singleSpread.Expression.Type is not null);
+
+            if (!ShouldUseAddRangeOrToListMethod(singleSpread.Expression.Type, toListOfElementType.Parameters[0].Type, singleSpread.EnumeratorInfoOpt?.GetEnumeratorInfo.Method, listElementType))
+            {
+                return false;
+            }
+
+            result = _factory.Call(receiver: null, toListOfElementType, singleSpread.Expression);
+            return true;
+        }
+
+        private bool ShouldUseAddRangeOrToListMethod(TypeSymbol spreadType, TypeSymbol targetEnumerableType, MethodSymbol? getEnumeratorMethod, TypeWithAnnotations elementType)
```

Comment:
>TypeWithAnnotations elementType [](http://example.com/codeflow?start=144&length=31)

Minor: The parameter `elementType` seems redundant. We only use that parameter to determine whether the enumerable `targetEnumerableType` is an `ICollection<TElement>` before calling `Enumerable.ToList<TElement>(IEnumerable<TElement>)` or `List<TElement>.AddRange(IEnumerable<TElement>)`. But we should use the type argument from the enumerable type directly since that is the `IEnumerable<TElement>` we're using to call those methods. (It looks like `elementType` and the type argument to `targetEnumerableType` are the same in the two callers, but it might be clearer if we avoid the extra parameter.)

In short, we could use the following below:
```
var iCollectionOfElementType = iCollectionOfTType.Construct(
    ((NamedTypeSymbol)targetEnumerableType).TypeArgumentsWithAnnotationsNoUseSiteDiagnostics);
```